Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs

```diff
@@ -106,7 +110,48 @@ buildRideListItem QRide.RideItem {..} = do
       }
 
 ---------------------------------------------------------------------------------------------------
+ticketRideList :: ShortId DM.Merchant -> Maybe (ShortId Common.Ride) -> Maybe Text -> Maybe Text -> Maybe Text -> Flow Common.TicketRideListRes
+ticketRideList merchantShortId mbRideShortId countryCode mbPhoneNumber _ = do
+  merchant <- findMerchantByShortId merchantShortId
+  let totalRides = 5
+  let mbShortId = coerce @(ShortId Common.Ride) @(ShortId DRide.Ride) <$> mbRideShortId
+  case mbPhoneNumber of
+    Just number -> do
+      no <- getDbHash number
+      let code = fromMaybe "+91" countryCode
+      person <- runInReplica $ QPerson.findByMobileNumberAndMerchant code no merchant.id >>= fromMaybeM (PersonWithPhoneNotFound number)
+      allRides <- QRide.findAllRidesByDriverId person.id
+      let sortedRides = sortBy (compare `on` (.createdAt)) allRides
```

Comment:
Why do we need to sort?

Reply:
`findAllRidesByDriverId` returns all the rides but we want last 5 rides so sorting them on the basis of time at which ride is created and just taking last 5 of them

Reply:
Should we write Query that does Order By and Limits 5

Reply:
Same logic will be there but just the change will that it will be a query as in query also we need to sort and give 5 rides back